hudi-client/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -61,6 +61,7 @@
   private Map<String, HoodieRecord<T>> keyToNewRecords;
   private Set<String> writtenRecordKeys;
   private HoodieStorageWriter<IndexedRecord> storageWriter;
+  private String partitionPath;
```

Comment:
Now that all handles have partitionPath, can this be moved to base class (HoodieWriteHandle ?)

Reply:
Fixed. I tried to see if there is a way to have this partition validation check in one place, but couldn't find a common entry point. Please let me know if you have suggestions here. happy to refactor more if needed